test/unit_test/cb_large_actions_test.cc

```diff
@@ -79,192 +79,369 @@ BOOST_AUTO_TEST_CASE(creation_of_the_og_A_matrix)
+
+      vw.learn(examples);
+      vw.finish_example(examples);
+    }
 
-  {
-    VW::multi_ex examples;
+    {
+      VW::multi_ex examples;
+
+      examples.push_back(VW::read_example(vw, "| 1 2 3"));
+      examples.push_back(VW::read_example(vw, "| a_1 a_2 a_3"));
+      examples.push_back(VW::read_example(vw, "| a_4 a_5 a_6"));
+      examples.push_back(VW::read_example(vw, "0:0.7:0.5 | a_7 a_8 a_9"));
+      examples.push_back(VW::read_example(vw, "| a_10 a_11 a_12"));
+      examples.push_back(VW::read_example(vw, "| a_13 a_14 a_15"));
+      examples.push_back(VW::read_example(vw, "| a_16 a_17 a_18"));
+
+      vw.learn(examples);
+      vw.finish_example(examples);
+    }
 
-    examples.push_back(VW::read_example(vw, "0:1.0:0.5 | 1 2 3"));
-    examples.push_back(VW::read_example(vw, "| a_1 a_2 a_3"));
+    {
+      VW::multi_ex examples;
+
+      examples.push_back(VW::read_example(vw, "| 1 2 3"));
+      examples.push_back(VW::read_example(vw, "| a_1 a_2 a_3"));
+      examples.push_back(VW::read_example(vw, "| a_4 a_5 a_6"));
+      examples.push_back(VW::read_example(vw, "| a_7 a_8 a_9"));
+      examples.push_back(VW::read_example(vw, "0:0.6:0.5 | a_10 a_11 a_12"));
+      examples.push_back(VW::read_example(vw, "| a_13 a_14 a_15"));
+      examples.push_back(VW::read_example(vw, "| a_16 a_17 a_18"));
+
+      vw.learn(examples);
+      vw.finish_example(examples);
+    }
 
-    vw.predict(examples);
+    {
+      VW::multi_ex examples;
+
+      examples.push_back(VW::read_example(vw, "| 1 2 3"));
+      examples.push_back(VW::read_example(vw, "| a_1 a_2 a_3"));
+      examples.push_back(VW::read_example(vw, "| a_4 a_5 a_6"));
+      examples.push_back(VW::read_example(vw, "| a_7 a_8 a_9"));
+      examples.push_back(VW::read_example(vw, "| a_10 a_11 a_12"));
+      examples.push_back(VW::read_example(vw, "0:0.5:0.5 | a_13 a_14 a_15"));
+      examples.push_back(VW::read_example(vw, "| a_16 a_17 a_18"));
+
+      vw.learn(examples);
+      vw.finish_example(examples);
+    }
 
-    action_space->explore._generate_A(examples);
-    action_space->explore.generate_Y(examples);
+    {
+      VW::multi_ex examples;
+
+      examples.push_back(VW::read_example(vw, "| 1 2 3"));
+      examples.push_back(VW::read_example(vw, "| a_1 a_2 a_3"));
+      examples.push_back(VW::read_example(vw, "| a_4 a_5 a_6"));
+      examples.push_back(VW::read_example(vw, "| a_7 a_8 a_9"));
+      examples.push_back(VW::read_example(vw, "| a_10 a_11 a_12"));
+      examples.push_back(VW::read_example(vw, "| a_13 a_14 a_15"));
+      examples.push_back(VW::read_example(vw, "0:0.4:0.5 | a_16 a_17 a_18"));
+
+      vw.learn(examples);
+      vw.finish_example(examples);
+    }
+
+    std::vector<std::string> e_r;
+    vw.l->get_enabled_reductions(e_r);
+    if (std::find(e_r.begin(), e_r.end(), "cb_explore_adf_large_action_space") == e_r.end())
+    { BOOST_FAIL("cb_explore_adf_large_action_space not found in enabled reductions"); }
```

Comment:
get_learner_by_name throws if its not found, so maybe it can be minimized to BOOST_CHECK_NO_THROW ( ) but not sure if you can have an assignment inside that boost check function. also a nit, feel free to ignore